text/experimental/remote-sampling.md

```diff
@@ -0,0 +1,146 @@
+  // see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration
+  string name = 1; // TODO this can be enum (always_on, always_off, parentbased_traceidratio...)
+  // Sampler argument, equivalent to OTEL_TRACES_SAMPLER_ARG. Example 0.25 for traceidratio sampler.
+  string arg = 2;
+}
+
+// Request to get sampling configuration.
+message RemoteSamplingRequest {
+  string service_name = 1;
+  // TODO shall we add version and/or environment?
+}
+
+// Service that is used to get sampling configuration from collector.
+service SamplingManager {
+  // Get sampling configuration for a given workload.
+  rpc GetSamplingStrategy(RemoteSamplingRequest) returns (ServiceSamplingResponse) {}
+}
+```
+
+Additional SDK requirements:
+
+* Add `RemoteSampler` to [trace/sdk.md](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#built-in-samplers)
+* Add `remote` to `OTEL_TRACES_SAMPLER` and `poolingIntervalMs`, `initialSampler`, `initialSamplerArg` to `OTEL_TRACES_SAMPLER_ARG` [sdk-environment-variables.md](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration)
+
+Remote Sampler configuration:
+
+* pooling interval - pooling interval for getting configuration from remote
+* endpoint - address of running service with sampling manager
+* initial sampler - initial sampler that is used before the first configuration is fetched
+
+## Trade-offs and mitigations
+
+## Prior art and alternatives
+
+Jaeger client libraries support [remotely controlled sampling](https://www.jaegertracing.io/docs/1.23/sampling/).
+
+The sampling configuration is divided into two parts: service and default strategy.
+The service strategy defines sampling at a service or operation level. The default strategy is
+used when no service was matched.
+
+Follows an example JSON configuration for the Jaeger Collector:
+
+```json
+{
+  "service_strategies": [
+    {
+      "service": "foo",
+      "type": "probabilistic",
+      "param": 0.8,
+      "operation_strategies": [
+        {
+          "operation": "op1",
+          "type": "probabilistic",
+          "param": 0.2
+        },
+        {
+          "operation": "op2",
+          "type": "probabilistic",
+          "param": 0.4
+        }
+      ]
+    },
+    {
+      "service": "bar",
+      "type": "ratelimiting",
+      "param": 5
+    }
+  ],
+  "default_strategy": {
+    "type": "probabilistic",
+    "param": 0.5,
+    "operation_strategies": [
+      {
+        "operation": "/health",
+        "type": "probabilistic",
+        "param": 0.0
+      },
+      {
+        "operation": "/metrics",
+        "type": "probabilistic",
+        "param": 0.0
+      }
+    ]
+  }
+}
+```
+
+## Open questions
+
+* Should we support regex match on the span name or attributes? It can be useful for matching `http.url` attribute.
```

Comment:
Yes :) Also for matching against Baggage.

Reply:
I would be more careful with baggage. The sampler API does not explicitly take it https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#span-creation. However I know it can be extracted from the context.

I would prefer to add regex support in the second iteration.